hydrogym/firedrake/envs/cylinder/flow.py

```diff
@@ -149,35 +155,25 @@ def shear_force(self, q: fd.Function = None) -> float:
         (direction / self.Re * sqrt(du_dn_t[0]**2 + du_dn_t[1]**2)) *
         ds(self.CYLINDER))
 
-  # TODO: Add back in when linearization is fixed
-  # def linearize_control(self, qB=None):
-  #     """
-  #     Solve linear problem with nonzero Dirichlet BCs to derive forcing term for unsteady DNS
-  #     """
-  #     if qB is None:
-  #         qB = self.solve_steady()
-
-  #     A = self.linearize_dynamics(qB, adjoint=False)
-  #     # M = self.mass_matrix()
-  #     self.linearize_bcs()  # Linearize BCs first (sets freestream to zero)
-  #     self.set_control([1.0])  # Now change the cylinder rotation  TODO: FIX
-
-  #     (v, _) = fd.TestFunctions(self.mixed_space)
-  #     zero = fd.inner(fd.Constant((0, 0)), v) * fd.dx  # Zero RHS for linear form
-
-  #     f = fd.Function(self.mixed_space)
-  #     fd.solve(A == zero, f, bcs=self.collect_bcs())
-  #     return f
-
   def linearize_bcs(self, function_spaces=None):
     self.reset_controls(function_spaces=function_spaces)
     self.bcu_inflow.set_value(fd.Constant((0, 0)))
     self.bcu_freestream.set_value(fd.Constant(0.0))
 
-  def evaluate_objective(self, q: fd.Function = None) -> float:
+  def evaluate_objective(self,
+                         q: fd.Function = None,
+                         averaged_objective_values=None,
+                         lambda_value=0.2,
+                         return_objective_values=False) -> float:
```

Comment:
- What's the purpose of `return_objective_values`?  This changes the return type and makes the behavior of the function fairly unclear.  Also does not seem to be supported by all environments.  I think this should probably be removed.
- I would also rename `lambda_value` to something like `CL_weight` to be more specific.
- If `averaged_objective_values` is supposed to be a general part of the interface then it should be part of the signature in `PDEBase.evaluate_objective`

Reply:
Indeed, l. 171 could be move below to separate the `CL`, `CD` quantity computation vs. the cost derivation type